### ISSUE TITLE
Objects create an unexpected way of recurring

### ISSUE THREAD

**Issue by 71104** [71104/lambda#106]
```lambda
let Math.fibonacci = fn Math, n ->
  if n < 2
  then 1
  else Math.fibonacci (n - 2) + Math.fibonacci (n - 1)
in Math.fibonacci 5
```

Is this correct/acceptable?

**Comment by 71104** [comment]
This actually fixes #99 pretty well.

**Comment by 71104** [comment]
Also, the type system should prevent this because `Math.fibonacci` can't be typed inside itself.
